### PR TITLE
fix(classifier): skip label validation for bat embedding model

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- Fix bat classifier failing to load when the user has a custom BirdNET model with a different species count than the standard v2.4 (6522 classes)
- The bat pipeline uses BirdNET v2.4 embeddings extraction followed by a custom bat classifier head; the embedding model's labels are unused (only embeddings are extracted), but label count validation blocked initialization
- Add `WithSkipLabelValidation` option to skip the label-vs-model-output check for embedding-only use
- Decouple output tensor allocation from label count by adding `LogitsSize` to `ModelConfig`, populated from actual ONNX model metadata, preventing tensor shape mismatches at inference time

## Test Plan

- [ ] Deploy to production instance with custom BirdNET model (6559 labels) and verify bat classifier loads
- [ ] Verify bat classifier appears in inference timing stats
- [ ] Verify standard BirdNET and Perch v2 models are unaffected (label validation still enforced)
- [ ] `go build` and `golangci-lint run -v` pass

## Summary by CodeRabbit

* **New Features**
  * Added an option to skip label-count validation when loading models, allowing embedding extraction even if provided labels don’t match the model’s logits dimension.

* **Improvements**
  * Model metadata now records the actual logits output dimension, improving compatibility and correctness when interpreting model outputs.

[Review Change Stack](https://app.coderabbit.ai/change-stack/tphakala/birdnet-go/pull/3051)

### CODE REVIEW

File: internal/inference/onnx/classifier.go [hunk 1]
```diff
@@ -275,12 +285,12 @@ func (c *Classifier) PredictRawWithEmbeddings(audio []float32) (logits, embeddin
...
-	if numClasses > len(allLogits) {
-		return nil, nil, fmt.Errorf("birdnet: logits tensor too small: need %d, have %d", numClasses, len(allLogits))
+	logitsSize := c.config.LogitsSize
+	if logitsSize > len(allLogits) {
+		return nil, nil, fmt.Errorf("birdnet: logits tensor too small: need %d, have %d", logitsSize, len(allLogits))
 	}
-	logits = make([]float32, numClasses)
-	copy(logits, allLogits[:numClasses])
+	logits = make([]float32, logitsSize)
+	copy(logits, allLogits[:logitsSize])
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/onnx/classifier.go:296]: Fixed in 679f28e2 - added `<= 0` guard in both `PredictRawWithEmbeddings` and `processOutput`.

File: internal/inference/onnx/detection.go [hunk 2]
```diff
@@ -68,10 +69,12 @@ func buildModelConfig(mt ModelType, inputShape []int64, outputShapes [][]int64)
 		}
 	case BirdNETv30:
 		cfg.LogitsIndex = 1
+		cfg.LogitsSize = lastDim(outputShapes[1])
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/onnx/detection.go:72]: The existing code already accesses `outputShapes` at hardcoded indices without bounds checks for other fields (`EmbeddingSize`, `EmbeddingIndex`) in the same switch cases. Model type detection in `resolveModelType` guarantees the output count matches the detected type, so these indices are always valid. Adding guards only to `LogitsSize` lines would be inconsistent with the existing pattern.

File: internal/inference/onnx/classifier.go [hunk 3]
```diff
@@ -106,8 +114,10 @@ func NewClassifier(modelPath string, opts ...ClassifierOption) (*Classifier, err
 		return nil, err
 	}
 
-	if err := validateLabelCount(&modelCfg, outputInfos, len(labels)); err != nil {
-		return nil, err
+	if !cfg.skipLabelValidation {
+		if err := validateLabelCount(&modelCfg, outputInfos, len(labels)); err != nil {
+			return nil, err
+		}
 	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/onnx/classifier.go:121]: When `skipLabelValidation` is false, the existing `validateLabelCount` call at line 116 already ensures labels match the model output at construction time. When true, the mismatch is intentional (embedding-only use). The `topK` function in `processOutput` handles length differences via `min(len(scores), len(labels))`, so there's no runtime risk from the mismatch.
